### PR TITLE
Rk cleanup

### PR DESCRIPTION
**Pull request by regiskuckaertz**
## Why are you doing this?

This is a routine cleanup task. There is no change in the logic or the UI the new contributions components yield, but there has been several health-related comments in my previous PRs that I wanted to address once and for all.

I decided to add `Object.value` and `Object.entries` because they are widely available and the polyfills weigh nothing at all.

For reviewing, it is better to look at each commit in isolation; they should be self-explanatory.

[**Trello Card**](https://trello.com/c/zRSxRltu/789-npf-class-names-bem-data-in-currencyjs-contributionsjs-objectkeys)

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
❤️  mucho appreciado  😄

File: assets/helpers/internationalisation/contributions.js [hunk 1]
```diff
@@ -21,86 +19,30 @@ const countryGroupSpecificDetails: {
...
-    currency: {
-      name: 'Canadian Dollars',
-      symbol: '$',
-    },
-    contribution: {
-      oneoff: 'One-time',
-      monthly: 'Monthly',
-    },
   },
 };
```
Comment (Ap0c) [inline assets/helpers/internationalisation/contributions.js]: Would it make sense for this to live in `helpers/contributions`? Or would it make sense for some of the internationalisation logic there to live here? @joelochlann @Mullefa @tsop14 @jranks123

File: assets/helpers/internationalisation/currency.js [hunk 2]
```diff
@@ -48,7 +53,9 @@ const currencies: {
   },
 };
 
-const spokenCurrencies = {
+const spokenCurrencies: {
+  [IsoCurrency]: SpokenCurrency,
```
Comment (Ap0c) [inline assets/helpers/internationalisation/currency.js]: 👍

File: assets/pages/new-contributions-landing/components/ContributionAmount.jsx [hunk 3]
```diff
@@ -7,64 +7,74 @@ import { connect } from 'react-redux';
...
+import { classNameWithModifiers } from 'helpers/utilities';
 
 import SvgDollar from 'components/svgs/dollar';
 
+import { type State } from '../contributionsLandingReducer';
 
 // ----- Types ----- //
 
 type PropTypes = {
-  countryGroupDetails: CountryMetaData,
```
Comment (Ap0c) [inline assets/pages/new-contributions-landing/components/ContributionAmount.jsx]: You're still passing this through as a prop aren't you?

File: assets/pages/new-contributions-landing/components/ContributionState.jsx [hunk 4]
```diff
@@ -7,23 +7,30 @@ import { connect } from 'react-redux';
...
+import { type State } from '../contributionsLandingReducer';
 
 // ----- Types ----- //
 type PropTypes = {
   countryGroupId: CountryGroupId,
   selectedState: string,
 };
 
+const mapStateToProps: State => PropTypes = () => ({
+  countryGroupId: 'UnitedStates',
```
Comment (Ap0c) [inline assets/pages/new-contributions-landing/components/ContributionState.jsx]: Is this being double-passed? It's also being passed in as a prop at the component call-site in `contributionsLanding.jsx`.

File: assets/pages/new-contributions-landing/components/ContributionSubmit.jsx [hunk 5]
```diff
@@ -25,11 +26,13 @@ type PropTypes = {
 // ----- Render ----- //
 
 function ContributionSubmit(props: PropTypes) {
+  const currency: IsoCurrency = detect(props.countryGroupId);
```
Comment (Ap0c) [inline assets/pages/new-contributions-landing/components/ContributionSubmit.jsx]: This introduces a side-effect into this component (query param lookup), is it possible to use `fromCountryGroupId` from `currency.js` instead?

File: assets/pages/new-contributions-landing/components/ContributionAmount.jsx (hunk 3 above)
Comment (regiskuckaertz) [reply assets/pages/new-contributions-landing/components/ContributionAmount.jsx]: Good catch!

**Pull request merged**